File: Python/Examen-Parcial-1/Ejercicio1_lic_informatica.py
# Kevin Rafael Díaz López - 28/10/2024
# Programa que imprime en consola los números, separados por comas, del 1 hasta un número ingresado por el usuario.

# Códigos para dar color al título.
GREEN = "\033[32m"  # Define el color verde para el texto.
RESET = "\033[0m"  # Define el color de reinicio para volver al color original.

# Título del programa.
print(f"\n\t{GREEN}|LICENCIATURA EN INFORMÁRTICA|{RESET}")  # Imprime el título del programa en color verde.
print()  # Imprime una línea en blanco.

numero = int(input("Ingrese un número tope: "))  # Solicita un número entero.
i = 1  # Inicializa la variable "i" en 1, que será el contador para el bucle.

while i <= numero:  # Bucle que se ejecuta mientras "i" sea menor o igual a "numero".
    if i%3 == 0 and i%5 == 0: # Comprueba si es múltiplo de 3 o 5 (usando la lógica del módulo y el operador "and").
        print(end="Licenciatura en Informática, \n")
    elif i%3 == 0: # Comprueba si es múltiplo de 3.
        print(end = "Licenciatura, ")
    elif i%5 == 0: # Comprueba si es múltiplo de 5.
        print(end = "Informática, ")
    else: # Si no se cumplió ninguna condición, imprime el valor de "i".
        print(i , end = ", ")  # Imprime el valor de "i" en la misma línea, separado por una coma.
    i = i + 1  # Incrementa "i" en 1 en cada iteración para avanzar al siguiente número.

File: Java/Examen-Parcial-1/Ejercicio1_lic_informatica.java
private void botonCalcularMouseClicked(java.awt.event.MouseEvent evt) {                                           
        try {
            var numero = Integer.parseInt(campo1.getText()); // Obtiene el número ingresado en el campo1.
            StringBuilder salida = new StringBuilder();  // Para acumular el resultado completo.
            
            for (int i = 1; i <= numero; i++) {  // Bucle que se ejecuta mientras "i" sea menor o igual a "numero".
                if (i % 3 == 0 && i % 5 == 0) {  // Comprueba si es múltiplo de 3 y 5.
                    salida.append("Licenciatura en Informática, \n");
                } else if (i % 3 == 0) {  // Comprueba si es múltiplo de 3.
                    salida.append("Licenciatura, ");
                } else if (i % 5 == 0) {  // Comprueba si es múltiplo de 5.
                    salida.append("Informática, ");
                } else {  // Si no se cumplió ninguna condición, imprime el valor de "i".
                    salida.append(i).append(", ");
                }
            }
            
            resultado.setText(salida.toString());  // Asigna el resultado acumulado al campo de texto "resultado".
        } catch (Exception e) {
            campo1.setText("");
            System.out.println("error"); // Imprime un mensaje de error en la consola.
            resultado.setText("Syntax Error"); // En el campo resultado, muestra un mensaje de error.
            JOptionPane.showMessageDialog(null, "No es un número entero"); // Muestra una ventana flotante.
        }
    }
